dll/win32/imm32/imm.c

```diff
@@ -3735,11 +3784,31 @@ BOOL WINAPI ImmDisableTextFrameService(DWORD dwThreadId)
 /***********************************************************************
  *              ImmEnumInputContext(IMM32.@)
  */
-
 BOOL WINAPI ImmEnumInputContext(DWORD dwThreadId, IMCENUMPROC lpfn, LPARAM lParam)
 {
-    FIXME("Stub\n");
-    return FALSE;
+    HIMC *phList;
+    DWORD dwIndex, dwCount;
+    BOOL ret = TRUE;
+    HIMC hIMC;
+
+    TRACE("ImmEnumInputContext(%lu, %p, %p)\n", dwThreadId, lpfn, lParam);
+
+    dwCount = Imm32AllocAndBuildHimcList(dwThreadId, &phList);
+    if (!dwCount)
+        return FALSE;
+
+    for (dwIndex = 0; dwIndex < dwCount; ++dwIndex)
+    {
+        hIMC = phList[dwIndex];
+        if (!Imm32IsValidHimc(hIMC))
```

Comment:
Why wouldn't it be valid? that hIMC entry comes directly from that allocated phList by Imm32AllocAndBuildHimcList and initialized by Imm32BuildHimcList(), and so its contents must be valid by construction ?

Reply:
I deleted the check.